volatility3/framework/plugins/linux/mountinfo.py

```diff
@@ -0,0 +1,227 @@
+        vfsmnt = mnt.mnt
+        dentry = vfsmnt.get_mnt_root()
+
+        path_reversed = []
+        while dentry != fs_root.dentry or vfsmnt.vol.offset != fs_root.mnt:
+            if dentry == vfsmnt.get_mnt_root() or dentry.is_root():
+                parent = mnt.get_mnt_parent().dereference()
+                # Escaped?
+                if dentry != vfsmnt.get_mnt_root():
+                    return None
+
+                # Global root?
+                if mnt.vol.offset != parent.vol.offset:
+                    dentry = mnt.get_mnt_mountpoint()
+                    mnt = parent
+                    vfsmnt = mnt.mnt
+                    continue
+
+                return None
+
+            parent = dentry.d_parent
+            dname = dentry.d_name.name_as_str()
+            path_reversed.append(dname.strip("/"))
+            dentry = parent
+
+        path = "/" + "/".join(reversed(path_reversed))
+        return path
+
+    @classmethod
+    def get_mountinfo(cls, mnt, task) -> Union[None, Tuple[int, int, str, str, str, List[str], 
+                                                           List[str], str, str, List[str]]]:
+        """Extract various information about a mount point.
+        It mimics the Linux kernel show_mountinfo function.
+        """
+        mnt_root = mnt.get_mnt_root()
+        if not mnt_root:
+            return None
+
+        mnt_root_path = mnt_root.path()
+        superblock = mnt.get_mnt_sb()
+
+        mnt_id: int = mnt.mnt_id
+        parent_id: int = mnt.mnt_parent.mnt_id
+
+        st_dev = f"{superblock.major}:{superblock.minor}"
+
+        path_root = cls._do_get_path(mnt, task.fs.root)
+        if path_root is None:
+            return None
+
+        mnt_opts: List[str] = []
+        mnt_opts.append(mnt.get_flags_access())
+        mnt_opts.extend(mnt.get_flags_opts())
+
+        # Tagged fields
+        fields: List[str] = []
+        if mnt.is_shared():
+            fields.append(f"shared:{mnt.mnt_group_id}")
+
+        if mnt.is_slave():
+            master = mnt.mnt_master.mnt_group_id
+            fields.append(f"master:{master}")
+            dominating_id = mnt.get_dominating_id(task.fs.root)
+            if dominating_id and dominating_id != master:
+                fields.append(f"propagate_from:{dominating_id}")
+
+        if mnt.is_unbindable():
+            fields.append("unbindable")
+
+        mnt_type = superblock.get_type()
+
+        devname = mnt.get_devname()
+        if not devname:
+            devname = "none"
+
+        sb_opts: List[str] = []
+        sb_opts.append(superblock.get_flags_access())
+        sb_opts.extend(superblock.get_flags_opts())
+
+        return MountInfoData(mnt_id, parent_id, st_dev, mnt_root_path, path_root, mnt_opts, fields, 
+                             mnt_type, devname, sb_opts)
+
+    def _get_mnt_namespace_mountpoints(self, mnt_namespace):
+        mnt_type = self._get_symbol_fullname("mount")
```

Comment:
Maintaining state to use the symbol table is a little bit... it kinda feels bad.  I'm trying to think how I'd do this.  I think originally symbols were supposed to lookup without a symbol table name for conveniences, but I think that fell by the wayside.  As such, I'd still suggest creating the module from `self.config['kernel']` just so that if the this could be called without `_get_tasks_mountpoints` having been executed.  Not a major point, and not a blocker, but if you are going to add state, I'd feel safer if you ensure the variable is defined in `__init__`.